int64 x_min
int64 y_min
int64 x_max
int64 y_max
